test/integration/certs_test.go

```diff
@@ -0,0 +1,52 @@
+package integration_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/blocky/nitrite/internal"
+)
+
+func TestFetchingNitroCertProvider_Roots(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		if testing.Short() {
+			t.SkipNow()
+		}
+
+		// given
+		cp := internal.NewFetchingNitroCertProvider()
+
+		// when
+		got, err := cp.Roots()
+
+		// then
+		require.NoError(t, err)
+		assert.NotEmpty(t, got)
+	})
+}
+
+// TestEmbeddedAndFetchedCertsEqual tests that the certs fetched from AWS
+// and the certs embedded in the repository are the same. If they are not,
+// it may be possible the AWS has updated the Nitro Enclaves root certs.
```

Comment:
I think this test should be enough to check if something has changed on AWS end.

Notice that I'm not checking if the hash at https://docs.aws.amazon.com/enclaves/latest/user/verify-root.html#validation-process has changed. I'm assuming that if the zip changes, so will the hash, to which we don't have a direct download link though.